### PR TITLE
refactor(experimental): rename mapCodec to transformCodec

### PR DESCRIPTION
**Pull request by lorisleiva**
Currently we have the following codec functions:

```
// Transforms Codec<T> to Codec<U> by providing mapping functions between T and U.
mapCodec(codec, U => T, T => U);

// Creates a Codec<Map<K, V>> by providing a Codec<K> and a Codec<V>.
getMapCodec(keyCodec, valueCodec);
```

Whilst they do completely different things, their names are too close to each other.

This PR fixes this by renaming `mapCodec` to `transformCodec` — i.e. `/map(Encoder|Decoder|Codec)/transform$1/g`.

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2434?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2434** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2434?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2433** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2433?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2430** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2430?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2419** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2419?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lorisleiva and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/codecs-core/src/__typetests__/transform-codec-typetest.ts
Comment (buffalojoec) [inline packages/codecs-core/src/__typetests__/transform-codec-typetest.ts:1]: Ahh lame that your Git diff looks like this.

**Review by buffalojoec — APPROVED** [review]
"Transform" I think makes it more clear. Nice!

transformer

**Review by mcintyre94 — APPROVED** [review]
Nice, agree that this is clearer!

**Comment by lorisleiva** [comment]
### Merge activity

* **Apr 4, 12:47 PM EDT**: @lorisleiva started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2434).

**Pull request merged**